### PR TITLE
bundled translations: use the locale and add docs

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

**Comment by ogoffart** [comment]
I'd like to be able to link to the rust/c++ docs from the guide docs, and vice-versa, but i'm not sure how to do it.

File: docs/src/content/docs/concepts/development/localization.md [hunk 1]
```diff
@@ -222,16 +222,45 @@ int main()
...
+
+### Bundling with C++
+
+Set the `SLINT_BUNDLE_TRANSLATIONS` property in CMake:
+
+```cmake
+set_property(TARGET my_application PROPERTY SLINT_BUNDLE_TRANSLATIONS "${CMAKE_CURRENT_SOURCE_DIR}/lang")
+```
+
+the `<domain>` is the cmake target name.
```
Comment (tronical) [inline docs/src/content/docs/concepts/development/localization.md:254]: Similarly.... did you mean to use a more generic example path perhaps?

File: docs/src/content/docs/concepts/development/localization.md [hunk 2]
```diff
@@ -222,16 +222,45 @@ int main()
...
+
+### Selecting a Translation
+
+If the `std` feature isenabled when building Slint, Slint will try to detect the language based on the locale.
```
Comment (tronical) [inline docs/src/content/docs/concepts/development/localization.md:258]:
```suggestion
If you enable the `std` feature with Slint, language for translations is detected based on the locale.
```

File: docs/src/content/docs/concepts/development/localization.md [hunk 3]
```diff
@@ -222,16 +222,45 @@ int main()
...
+If one of the bundled language matches the selected locale, it will be used.
+
+Use the `slint::select_bundled_translation` to change thanslation at runtime.
```
Comment (tronical) [inline docs/src/content/docs/concepts/development/localization.md:261]:
```suggestion
Use the `slint::select_bundled_translation` function to change translations at runtime.
```

**Review by tronical — APPROVED** [review]

**Pull request merged**